addons/voice/org.openhab.voice.pico2wave/src/main/java/org/openhab/voice/picotts/internal/PicoTTSService.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.voice.picotts.internal;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.eclipse.smarthome.core.audio.AudioException;
+import org.eclipse.smarthome.core.audio.AudioFormat;
+import org.eclipse.smarthome.core.audio.AudioStream;
+import org.eclipse.smarthome.core.voice.TTSException;
+import org.eclipse.smarthome.core.voice.TTSService;
+import org.eclipse.smarthome.core.voice.Voice;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Florian Schmidt - Initial Contribution
+ */
+@Component
+public class PicoTTSService implements TTSService {
+    private final Set<Voice> voices = Arrays
+            .asList(new PicoTTSVoice("de-DE"), new PicoTTSVoice("en-US"), new PicoTTSVoice("en-GB"),
+                    new PicoTTSVoice("es-ES"), new PicoTTSVoice("fr-FR"), new PicoTTSVoice("it-IT"))
+            .stream().collect(Collectors.toSet());
+
+    private final Set<AudioFormat> audioFormats = Collections.singleton(
+            new AudioFormat(AudioFormat.CONTAINER_WAVE, AudioFormat.CODEC_PCM_SIGNED, false, 16, null, null));
+
+    @Override
+    public Set<Voice> getAvailableVoices() {
+        return this.voices;
+    }
+
+    @Override
+    public Set<AudioFormat> getSupportedFormats() {
+        return this.audioFormats;
+    }
+
+    @Override
+    public AudioStream synthesize(String text, Voice voice, AudioFormat requestedFormat) throws TTSException {
+        if (text == null || text.isEmpty()) {
+            throw new TTSException("The passed text can not be null or empty");
+        }
+
+        if (!this.voices.contains(voice)) {
+            throw new TTSException("The passed voice is no supported");

```

Comment:
I'd add a "t" so the message has  "not supported" or maybe use "unsupported" here as well since the exception message below also uses it.

Reply:
I missed the t while typing, however, unsupported sounds even better :)